eth/stagedsync/stage_execute.go

```diff
@@ -461,13 +461,13 @@ Loop:
 			if !errors.Is(err, context.Canceled) {
 				logger.Warn(fmt.Sprintf("[%s] Execution failed", logPrefix), "block", blockNum, "hash", block.Hash().String(), "err", err)
 				if cfg.hd != nil {
-					cfg.hd.ReportBadHeaderPoS(blockHash, block.ParentHash())
+					cfg.hd.ReportBadHeaderPoS(blockHash, block.ParentHash() /* lastValidAncestor */)
 				}
 				if cfg.badBlockHalt {
 					return err
 				}
+				u.UnwindTo(blockNum-1, blockHash /* badBlock */)
```

Comment:
Looking at the executeBlock code there are a number or reasons for failure, other than a block with an invalid hash.  Marking the block as bad via unwind will mean that a header with that has can never be delivered, even if it is a valid hash on the chain.  I think it will always cause erigon to halt processing as it won't be able to get past that header.

Reply:
The existing assumption here is that errors in block execution are deterministic/non-transient and it doesn't make sense to import a block that we can't execute. To my mind that a reasonable assumption and if it doesn't hold (`executeBlock` producing transient errors), that's a bug to be fixed.